components/ChallengeMaterial.tsx

```diff
@@ -25,6 +20,19 @@ type ReviewStatusProps = {
   reviewerUserName: string | null
 }
 
+export type ChallengeSubmissionData = {
+  title: string
+  id: string
+  order: number
+  description: string
+  status: string
+  submission?: Submission
+}
```

Comment:
I'd suggest changing this with a compound type since there's a lot of redundant stuff but it breaks some things here. If you wanna try to do it then great, but it's not needed in this PR.

```suggestion
export type ChallengeSubmissionData = Challenge & { submission: Submission }
```

Would probably be good to rename with with a better name too like `ChallengeWithSubmission`.